clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/ApplicationEventHandler.java

```diff
@@ -39,29 +43,36 @@
  * An event handler that receives {@link ApplicationEvent application events} from the application thread which
  * are then readable from the {@link ApplicationEventProcessor} in the {@link ConsumerNetworkThread network thread}.
  */
+@SuppressWarnings("this-escape")
 public class ApplicationEventHandler implements Closeable {
 
     private final Logger log;
+    private final Time time;
     private final BlockingQueue<ApplicationEvent> applicationEventQueue;
     private final ConsumerNetworkThread networkThread;
     private final IdempotentCloser closer = new IdempotentCloser();
+    private final Optional<KafkaAsyncConsumerMetrics> kafkaAsyncConsumerMetrics;
 
     public ApplicationEventHandler(final LogContext logContext,
                                    final Time time,
                                    final BlockingQueue<ApplicationEvent> applicationEventQueue,
                                    final CompletableEventReaper applicationEventReaper,
                                    final Supplier<ApplicationEventProcessor> applicationEventProcessorSupplier,
                                    final Supplier<NetworkClientDelegate> networkClientDelegateSupplier,
-                                   final Supplier<RequestManagers> requestManagersSupplier) {
+                                   final Supplier<RequestManagers> requestManagersSupplier,
+                                   final Optional<KafkaAsyncConsumerMetrics> kafkaAsyncConsumerMetrics) {
         this.log = logContext.logger(ApplicationEventHandler.class);
+        this.time = time;
         this.applicationEventQueue = applicationEventQueue;
+        this.kafkaAsyncConsumerMetrics = kafkaAsyncConsumerMetrics;
         this.networkThread = new ConsumerNetworkThread(logContext,
                 time,
-                applicationEventQueue,
+                this,
```

Comment:
I'm not crazy about this. (Haha, pun intended.)

With this change there's now a bidirectional relationship between the event handler and the background thread. I'd hope we could find a way to maintain the decoupled nature of the event queue from the event handler 🤔

Reply:
Revert the change. Create a followup Jira: https://issues.apache.org/jira/browse/KAFKA-18048.